test/yargs-parser.js

```diff
@@ -176,6 +176,20 @@ describe('yargs-parser', function () {
     x.should.be.a('string').and.equal('56')
   })
 
+  it('should default numbers to undefined', function () {
+    var n = parser([ '-n' ], {
+      number: ['n']
+    }).n
+    expect(n).to.equal(undefined)
```

Comment:
defaulting to `undefined` is better imo. Dealing with `NaN` has always been a pain in the butt in my experience.